texthero/embeddings.py

```diff
@@ -0,0 +1,123 @@
+"""
+embeddings.py
+-------------
+
+Embed documents and work with the embeddings to find similar documents,
+find topics, ... .
+
+There are many different ways to transform text data into vectors to gain
+insights from them. The resulting vectors are called _embeddings_.
+A _word embedding_ assigns a vector to each word, a _document embedding_
+(sometimes called thought _thought vector_)
+assigns a vector to each document.
+
+One way to get embeddings is to use a function such as tfidf,
+count, or term_frequency that creates vectors depending
+on which words occur how frequently.
+
+Another option is to use embeddings that try to directly capture
+the semantic relationship between words or sentences. For example,
+the vector of 'biochemistry' minus the vector of 'chemistry' is
+close to the vector of 'biology'.
+
+In Texthero, both options are supported. The second option is
+implemented through the `Flair library <https://github.com/flairNLP/flair>`_
+"""
+
+import pandas as pd
+import numpy as np
+
+# We only `import flair` and
+# call everything from flair directly,
+# e.g. flair.data.Tokens, to not pollute our
+# namespace with similarly sounding names.
+import flair
+
+from typing import List, Union
+
+from texthero._types import InputSeries, TokenSeries, VectorSeries
+
+
+"""
+Helper functions.
+"""
+
+
+def _texthero_init_for_flair_sentence(self, already_tokenized_text: List[str]):
+    """
+    To use Flair embeddings, Flair needs as input a
+    'flair.Sentence' object. Creating such an object
+    only works from strings in flair. However, we want
+    our embeddings to work on TokenSeries, so we
+    overwrite the 'flair Sentence' __init__ method with
+    this method to create Sentence objects from already
+    tokenized text.
+    """
+
+    super(flair.data.Sentence, self).__init__()
+
+    self.tokens: List[flair.data.Token] = []
+    self._embeddings: Dict = {}
+    self.language_code: str = None
+    self.tokenized = None
+
+    # already tokenized -> simply add the tokens
+    for token in already_tokenized_text:
+        self.add_token(token)
+
+
+# Overwrite flair Sentence __init__ method to handle already tokenized text
+flair.data.Sentence.__init__ = _texthero_init_for_flair_sentence
+
+
+"""
+Support for flair embeddings.
+"""
+
+
+@InputSeries(TokenSeries)
+def embed(
```

Comment:
The first idea was to have this under `representation`, i.e to keep the number of modules fix to four ... opinion?

Reply:
Our thinking was to move everything that _takes TokenSeries and produces a vector per text_, so `TokenSeries -> VectorSeries` and `TokenSeries -> RepresentationSeries` under one `embeddings.py` file, so we'd move `tfidf, count, term_frequency` there too after this is merged.

However, we could also put this into representation, probably just a question of how we see the trade-off _long files vs. many files_.

Reply:
I agree that we don't want very long files (but in my experience sometimes it's easier to find things in a long document with CTRL-F rather than search in multiple files and folders and understand a more complex structure). I also understand and share your same view on the trade-off.

What I would suggest is that we can split the representation file into two files but we hide this to users, i.e in the API all these functions will appear under `texthero.representation` (that's what I meant by saying to keep only 4 modules)

Reply:
Yes makes sense. Maybe I should move this `embeddings.py` stuff to `representation` for the time being and we can do the re-org in a separate PR?

Reply:
Yes!